lib/chef/provider/package/chocolatey.rb

```diff
@@ -136,14 +136,27 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||=
+          @choco_exe ||= begin
+            path = choco_install_path
+
+            # GetEnvironmentVariable returns "" on failure.
+            if path.to_s.empty?
+              raise Chef::Exceptions::MissingLibrary.new("Could not find Chocolatey. Is it installed?")
```

Comment:
oh I didn't read @danielsdeleo 's first note very carefully. So first I would add a note about running the [chocolatey cookbook](https://github.com/chocolatey/chocolatey-cookbook) to install it if we cant find it. Its officially supported under the chocolatey org and I'm its primary maintainer. 

Perhaps logging to debug that the Machine scoped `ChocolateyInstall` variable either was not present or was found to be X could suffice for troubleshooting purposes. If the user changes this variable after choco is installed and choco.exe is no longer found under that path, they have basically broken all chocolatey functionality. The more possibly risky scenario is that they install chocolatey and it is there, but the environment variable has not been propperly broadcast and is not seen by this resource. However the chocolatey installer should do that.